packages/polling-controller/src/BlockTrackerPollingController.ts

```diff
@@ -0,0 +1,87 @@
+import { BaseController, BaseControllerV1 } from '@metamask/base-controller';
+import type {
+  NetworkClientId,
+  NetworkClient,
+} from '@metamask/network-controller';
+import type { Json } from '@metamask/utils';
+
+import {
+  AbstractPollingControllerBaseMixin,
+  getKey,
+} from './AbstractPollingController';
+import type { PollingTokenSetId } from './AbstractPollingController';
+
+type Constructor = new (...args: any[]) => object;
+
+/**
+ * BlockTrackerPollingControllerMixin
+ * A polling controller that polls using a block tracker.
+ *
+ * @param Base - The base class to mix onto.
+ * @returns The composed class.
+ */
+function BlockTrackerPollingControllerMixin<TBase extends Constructor>(
+  Base: TBase,
+) {
+  abstract class BlockTrackerPollingController extends AbstractPollingControllerBaseMixin(
+    Base,
+  ) {
+    #activeListeners: Record<string, (options: Json) => Promise<void>> = {};
+
+    abstract _getNetworkClientById(
```

Comment:
This method is used just below on line 45 to retrieve the networkClient blockTracker. The StaticIntervalPollingController does not  require the networkClient. The reason I don't have this as a constructor arg is because safely adding a required param to the constructor for mixins is tricky and generally not recommended since the children will have diverse constructor signatures

Reply:
I see, okay.